File: src/main/java/com/elroykanye/sadues/business/service/i/MemberService.java
package com.elroykanye.sadues.business.service.i;

import com.elroykanye.sadues.api.dto.MemberDto;
import com.elroykanye.sadues.api.dto.response.SaResponse;
import com.elroykanye.sadues.data.entity.relation.Member;

import java.util.List;

public interface MemberService {
    
    SaResponse create( MemberDto dto);

    
    Member getEntity(MemberDto.MemberKeyDto keyDto);

    
    MemberDto getDto(MemberDto.MemberKeyDto keyDto);

    
    List<Member> getAllEntities();

    
    List<MemberDto> getAllDto();

    
    SaResponse update( MemberDto dto);

    void delete(MemberDto.MemberKeyDto keyDto);
}

File: src/main/java/com/elroykanye/sadues/business/service/i/MemberService.kt
package com.elroykanye.sadues.business.service.i

import com.elroykanye.sadues.api.dto.MemberDto
import com.elroykanye.sadues.api.dto.response.SaResponse
import com.elroykanye.sadues.data.entity.relation.Member

interface MemberService {
    fun create(memberDto: MemberDto): SaResponse

    fun getEntity(userId: Long, associationId: Long): Member

    fun getDto(userId: Long, associationId: Long): MemberDto

    fun getAllEntities(): List<Member>

    fun getAllDto(): List<MemberDto>

    fun update(memberDto: MemberDto): SaResponse

    fun delete(userId: Long, associationId: Long)

} 
